swfiles/@spinw/genmagstr.m

```diff
@@ -50,9 +50,9 @@ function genmagstr(obj, varargin)
 % 
 % `'mode'`
 % : Mode that determines how the magnetic structure is generated:
-%   * `'random'` (reads -)
-%      generates random zero-k magnetic structure.
-%   * `'direct'` (reads `S`, `n`, `k`)
+%   * `'random'` (reads `k`, `n`)
+%           generates random zero-k magnetic structure.
```

Comment:
My understanding of how `'random'` works is as follows:

```
%   * `'random'` (optionally reads `k`, `n`, `nExt`)
%           generates a random structure in the structural cell if no other
%           arguments are specified here or previously in this spinw object.
%           If `nExt` is specified all spins in the supercell are randomised.
%           If `k` (and optionally `n`) is specified the structure is then treated
%           as incommensurate (similarly to the `'helical'` mode but with
%           randomized spins in the first cell).
```

If you agree, could you change the docstring to this, please?

Reply:
Actually I'm a bit confused about how `n` works with `random`, it is stored in the imaginary part of `mag_str.F`, but it doesn't appear to have any affect on the actual magnetic structure displayed with `plot`. I can see the imaginary part is used in calculating a different output `n` in `magstr`, but it does it then actually get used anywhere?

Reply:
Right... it gets used only if `k` is not `[0 0 0]` (or `[1 1 1]` etc). If `k` is not specified (default is `[0 0 0]`) or is integral then you don't see the effect of `n`.

[As explained here](https://spinw.org/magneticStructurePresentation/#/magstr_theory3) if the basisvector `F` is real, and you have a non-integral `k` then the moments in each cell will be parallel to each other but their magnitude ("spin length") will vary with the periodicity of `k`. There _are_ physical materials which behave like this (often called "spin density waves" or "amplitude modulated spin structures"), but they were not systems which Sandor was working much on.

Instead if the basisvector `F` is complex and the imaginary component is perpendicular to the real component then you get a helical system - and these were the materials which Sandor was mostly working on, so he set it up so that the random mode will by default create one of these systems instead. So, the [last line of the "random" mode](https://github.com/SpinW/spinw/blob/f4d907f5e891c46342585980be5eac1c9289c129/swfiles/%40spinw/genmagstr.m#L373) makes the imaginary component perpendicular to both `n` and `S` using the cross product.

Maybe the following text makes things a bit clearer?

```
%   * `'random'` (optionally reads `k`, `n`, `nExt`)
%           generates a random structure in the structural cell if no other
%           arguments are specified here or previously in this spinw object.
%           If `nExt` is specified all spins in the supercell are randomised.
%           If `k` is specified a random helical structure with moments
%           perpendicular to `n` (default value: `[0 0 1]`) with the specified
%           `k` propagation vector is generated. (`n` is not otherwise used).
```

Reply:
Thanks that makes a lot more sense now! I'll update the docstring